### PR TITLE
vmalert: speed up state restore procedure on start

### PR DESCRIPTION
**Pull request by hagen1778**
Alerts state restore procedure has been changed to become asynchronous. It doesn't block groups start anymore which significantly improves vmalert's startup time. Instead, state restore is called by each group in their goroutines after the first rules evaluation.

While previously state restore attempt was made for all loaded alerting rules, now it is called only for alerts which became active after the first evaluation. This reduces the amount of API calls to the configured remote read URL.

This also means that `remoteRead.ignoreRestoreErrors` command-line flag becomes deprecated now and will have no effect if configured.

See relevant issue https://github.com/VictoriaMetrics/VictoriaMetrics/issues/2608

Signed-off-by: hagen1778 <roman@victoriametrics.com>

### CODE REVIEW

**Review by f41gh7 — APPROVED** [review]
LGTM

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,10 @@ The following tip changes can be tested by building VictoriaMetrics components f
 
 ## tip
 
+* FEATURE: [vmalert](https://docs.victoriametrics.com/vmalert.html): alerts state restore procedure was changed to become asynchronous. It doesn't block groups start anymore which significantly improves vmalert's startup time. 
+This also means that `remoteRead.ignoreRestoreErrors` command-line flag becomes deprecated now and will have no effect if configured.
```
Comment (valyala) [inline docs/CHANGELOG.md:19]:
```suggestion
This also means that `-remoteRead.ignoreRestoreErrors` command-line flag becomes deprecated now and will have no effect if configured. 
```

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -15,6 +15,10 @@ The following tip changes can be tested by building VictoriaMetrics components f
 
 ## tip
 
+* FEATURE: [vmalert](https://docs.victoriametrics.com/vmalert.html): alerts state restore procedure was changed to become asynchronous. It doesn't block groups start anymore which significantly improves vmalert's startup time. 
+This also means that `remoteRead.ignoreRestoreErrors` command-line flag becomes deprecated now and will have no effect if configured. 
+While previously state restore attempt was made for all loaded alerting rules, now it is called only for alerts which became active after the first evaluation. See relevant issue [here](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/2608)
```
Comment (valyala) [inline docs/CHANGELOG.md:20]:
```suggestion
While previously state restore attempt was made for all the loaded alerting rules, now it is called only for alerts which became active after the first evaluation. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/2608).
```

**Review by valyala — APPROVED** [review]

**Review by valyala — APPROVED** [review]

**Comment by valyala** [comment]
@hagen1778 , thanks for the pull request!

**Comment by valyala** [comment]
@hagen1778 , could you port this pull request to the enterprise branch?

**Pull request merged**